core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -974,8 +974,10 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
     * which has performance implications. For maximum performance, `evalMapChunk`
     * is available, however, with caveats.
     */
-  def evalMap[F2[x] >: F[x], O2](f: O => F2[O2]): Stream[F2, O2] =
-    flatMap(o => Stream.eval(f(o)))
+  def evalMap[F2[x] >: F[x], O2](f: O => F2[O2]): Stream[F2, O2] = {
+    def evalOut(o: O): Pull[F2, O2, Unit] = Pull.eval(f(o)).flatMap(Pull.output1)
+    Pull.flatMapOutput[F, F2, O, O2](underlying, evalOut).streamNoScope
```

Comment:
Not as part of this PR, but we should probably make `flatMapOutput` a (private?) method on `Pull`.

Reply:
The problem is that it only applies for a pull where the result type is `Unit`.

Reply:
You can require an evidence `R =:= Unit` to use the method.

Reply:
> You can require an evidence R =:= Unit to use the method.

I avoid evidence parameters, since both extension methods in Scala 3 or even implicit classes in Scala 2 are, IMHO, cleaner. We can try that.

Reply:
Personal taste :) extension methods are much harder on discoverability, I would only reach for them when it is actually impossible to add the method to the type (i.e. you do not control it). I think it is easy to discover a method defined on the type, and the evidence parameter makes it clear what its constraints are.

Reply:
I guess the reason is: extension methods have the benefit of giving up-front the type pattern (e.g. the binding of R to Unit) on which they apply, so it almost read as "if your pull has a return type unit, then define `foo`", rather than "we have method `foo` over the list of type parameters, value parameters, and implicit parameters, provided that...". My personal taste or sense of style is to put conditions first, and in this case the 

Likewise, the extension class (already exists) looks to me as "for all Pulls such that their result type is Unit, let there be defined such methods as..."